sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathAccessControlEntry.cs

```diff
@@ -0,0 +1,233 @@
+
+        /// <summary>
+        /// Specifies which role this entry targets.
+        /// </summary>
+        public AccessControlType AccessControlType { get; set; }
+
+        /// <summary>
+        /// Specifies the entity for which this entry applies.
+        /// Must be omitted for types mask or other.  It must also be omitted when the user or group is the owner.
+        /// </summary>
+        public string EntityId { get; set; }
+
+        /// <summary>
+        /// Specifies the permissions granted to this entry.
+        /// </summary>
+        public RolePermissions Permissions { get; set; }
+
+        /// <summary>
+        /// Empty constructor.
+        /// </summary>
+        public PathAccessControlEntry() { }
+
+        /// <summary>
+        /// Constructor.
+        /// </summary>
+        /// <param name="accessControlType">Specifies which role this entry targets.</param>
+        /// <param name="permissions">Specifies the permissions granted to this entry.</param>
+        /// <param name="defaultScope">Indicates whether this is the default entry for the ACL.</param>
+        /// <param name="entityId">Optional entity ID to which this entry applies.</param>
+        public PathAccessControlEntry(
+            AccessControlType accessControlType,
+            RolePermissions permissions,
+            bool defaultScope = false,
+            string entityId = default)
+        {
+            if (entityId != null
+                && !(accessControlType == AccessControlType.User || accessControlType == AccessControlType.Group))
+            {
+                throw new ArgumentException("AccessControlType must be User or Group if entityId is specified.");
+            }
+
+            DefaultScope = defaultScope;
+            AccessControlType = accessControlType;
+            EntityId = entityId;
+            Permissions = permissions;
+        }
+
+        /// <summary>
+        /// Override of ToString().
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            StringBuilder stringBuilder = new StringBuilder();
+
+            if (DefaultScope)
+            {
+                stringBuilder.Append("default:");
+            }
+            stringBuilder.Append(AccessControlType.ToString().ToLowerInvariant());
+            stringBuilder.Append(":");
+            stringBuilder.Append(EntityId ?? "");
+            stringBuilder.Append(":");
+            stringBuilder.Append(Permissions.ToSymbolicString());
+
+            return stringBuilder.ToString();
+        }
+
+        /// <summary>
+        /// Parses the provided string into a <see cref="PathAccessControlEntry"/>
+        /// </summary>
+        /// <param name="aclString">The string representation of the access control list.</param>
+        /// <returns>A <see cref="PathAccessControlEntry"/>.</returns>
+        public static PathAccessControlEntry Parse(string aclString)
```

Comment:
In .NET we usually name this parameter just `s` (see `int.Parse` for an example).

Reply:
fixed.

Reply:
Most of these still have different names.

Reply:
Also renamed the string parameters in PathAccessControlEntry.